src/main/java/uk/gov/ons/census/casesvc/utility/QidCreator.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.ons.census.casesvc.utility;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class QidCreator {
+
+  @Value("${qid.modulus}")
+  private int modulus;
+
+  @Value("${qid.factor}")
+  private int factor;
+
+  public long createQid(int questionnaireType, int trancheIdentifier, long uniqueNumber) {
+    String sourceDigits =
+        String.format("%02d%01d%011d", questionnaireType, trancheIdentifier, uniqueNumber);
+
+    int checkDigits = calculateCheckDigits(sourceDigits);
```

Comment:
should we put a check in here, just in case the checkDigits are > 100 and then throw a runtime exception?